main_f7.c

```diff
@@ -764,14 +785,13 @@ main(void)
 #if defined(BOARD_USB_VBUS_SENSE_DISABLED)
 	try_boot = false;
 #else
-
-	if (gpio_get(GPIOA, GPIO9) != 0) {
+	if (board_test_usb_vbus()) {
```

Comment:
should this just be the assignment?
` try_boot = board_test_usb_vbus();`

and move the 'if defied(BOARD_USB_VBUS_SENSE_DISABLED)` to the function board_test_usb_vbus() so that it returns false when BOARD_USB_VBUS_SENSE_DISABLED is defined.

Reply:
Will do.

Reply:
We will have to be careful of doing `try_boot = board_test_usb_vbus();`, since try_boot is simply a flag that can be set to `false` by a number of different tests (receiving a UART break, if boot pin is strapped, if USB present). If we do the check `try_boot = board_test_usb_vbus(),` it would overwrite the previous check, which we don't want. We simply want a Or, check where if any of the checks returns true, then don't boot.